cmd/eksctl/logger.go

```
@@ -101,7 +101,7 @@ func dumpLogsToDisk(logBuffer *bytes.Buffer, errorString string) error {
 	if _, err := os.Stat("logs/"); os.IsNotExist(err) {
 
 		if err := os.Mkdir("logs/", 0755); err != nil {
-			return fmt.Errorf(err.Error())
+			return fmt.Errorf("%s", err.Error())
```

qq: why not just return the err? and same question for similar changes below

---

I was trying to stay light touch. The linter was complaining about no string format on the Errorf call so I just gave it one. The description of my original PR (that I had to cancel) has much better detail:

### Description
Adding support for new isolated AWS regions to include necessary changes for testing and fixing linter errors/complaints. A prototype binary was generated with these changes and we are able to successfully test functionality in region.

Key Changes For Build:
- pkg/apis/eksctl.io/v1alpha5/partitions.go: added
  - added logic to account for the four service endpoints that have non-standard domain prefixes in these partitions
  - partitions strings
  - partitions objects with region-appropriate
    - service mappings 
    - domain prefixes
  - added partitions to ISO part of GetEndpointServiceDomainPrefix 
- pkg/apis/eksctl.io/v1alpha5/types.go: added
  - in-region resource account IDs
  - region strings
  - SupportedRegions strings

Key Changes For Testing:
- pkg/actions/nodegroup/testdata all *.json files updated ServicePrincipalPartitionMap
- pkg/cfn/builder/karpenter_test.go updated expectedTemplate* template strings
- pkg/cfn/builder/testdata/nodegroup_access_entry all *.json files  ServicePrincipalPartitionMap
- pkg/cfn/builder/testdata/ added service_details_isob.json, service_details_isoe.json, service_details_isof.json
- pkg/cfn/builder/vpc_endpoint_test.go to add tests for isob, isoe and isof

Documentation:
- added new regions' information to 
  - userdocs/theme/home.html 
  - userdocs/src/getting-started.md

Linter Warnings/Errors:
- integration/tests/crud/creategetdelete_test.go: error and return value check for Sscanf
- added static string formatting to
  - cmd/eksctl/logger.go
  - pkg/actions/accessentry/task.go
  - pkg/actions/addon/tasks.go
  - pkg/actions/anywhere/anywhere.go
  - pkg/actions/podidentityassociation/tasks.go
  - pkg/cfn/template/matchers/matchers.go
  - pkg/eks/api_test.go
  - pkg/ssh/client/ssh.go
- unnecessary nil checks for 0 length slice
  - pkg/apis/eksctl.io/v1alpha5/addon.go
  - pkg/apis/eksctl.io/v1alpha5/addon.go